external-resolver/src/main/scala/org/genivi/sota/resolver/components/ComponentDirectives.scala

```diff
@@ -21,35 +21,39 @@ import Directives._
 
 /**
  * API routes for creating, deleting, and listing components.
- * @see {@linktourl http://pdxostc.github.io/rvi_sota_server/dev/api.html} 
+ * @see {@linktourl http://pdxostc.github.io/rvi_sota_server/dev/api.html}
  */
 class ComponentDirectives(implicit db: Database, mat: ActorMaterializer, ec: ExecutionContext) {
 
+  def searchComponent =
+    (get & parameter('regex.as[Refined[String, Regex]].?)) { re =>
```

Comment:
this is what this commit does as well:  https://github.com/advancedtelematic/rvi_sota_server/pull/328/commits/d432f8952a8d95f9e3a3ceeb12d2cab5ce346903